app/services/create_requested_account.rb

```diff
@@ -26,6 +26,12 @@ def initialize(requested_account, creator)
     process_request(@creator, @creation_reason)
   end
 
+  def self.format_failure_requested_account_result(result)
```

Comment:
This should be done as an instance method rather than a class method, so you can call something like `creation_attempt.result_description` to get the message that will be shown on the frontend. That will let you simplify the code in each of the controllers as well, since either way, `creation_attempt.result_description` will provide the message to show and you'll just need to use a conditional to determine whether to do the `JoinCourse` part.

Reply:
Done!